### PR TITLE
Improve ninja concurrency

### PR DESCRIPTION
**Pull request by fsaintjacques**
Until https://github.com/ninja-build/ninja/issues/1278 propagates in the upstream distribution, we need to limit the number of concurrent workers.

### CODE REVIEW

**Comment by kszucs** [comment]
That command would be run on the master side. Use properties instead.

Add a configuration value to the [yml](https://github.com/ursa-labs/ursabot/blob/master/prod.yaml#L57) and pass ncpus/ncores as a [property](https://github.com/buildbot/buildbot/blob/e7ff2a3b959cff96c77c07891fa07a35a98e81cb/master/buildbot/worker/base.py#L74) to the [DockerLatentWorker](https://github.com/ursa-labs/ursabot/blob/master/master.cfg#L145)

**Comment by fsaintjacques** [comment]
@kszucs if you can give a look, I didn't get a chance to validate it, my local workers don't behave :(

File: ursabot/builders.py [hunk 1]
```diff
@@ -325,7 +326,19 @@ def builders_for(cls, workers, images=tuple(), **kwargs):
     generator='Ninja',
     definitions=definitions
 )
-cpp_compile = Ninja(name='Compile C++', workdir='cpp/build')
+
+
+@util.renderer
+def ninja_concurrency(props):
+    n_cpus = len(os.sched_getaffinitiy(0))
```
Comment (kszucs) [inline ursabot/builders.py]: This runs on the master side.

Comment (fsaintjacques) [reply ursabot/builders.py]: Ok, I'll switch it to a `None` otherwise, and branch the argument accordingly.

**Comment by kszucs** [comment]
@fsaintjacques I'd prefer setting `ncpus` explicitly in the worker configuration instead of parsing it from the docker hostconfig. We'll have non-docker workers in the future.

**Comment by kszucs** [comment]
Closing in favor #130 (we can still reuse [counr_cpuspec](https://github.com/ursa-labs/ursabot/pull/127/files#diff-d855852af2b597bc44ce5148167e70f3R49) later)

**Pull request closed without merging**